cmd/influx/config.go

```diff
@@ -0,0 +1,196 @@
+	"github.com/spf13/cobra"
+)
+
+func cmdConfig(f *globalFlags, opt genericCLIOpts) *cobra.Command {
+	path, dir, err := defaultConfigPath()
+	if err != nil {
+		panic(err)
+	}
+	builder := cmdConfigBuilder{
+		genericCLIOpts: opt,
+		globalFlags:    f,
+		svc: config.LocalConfigsSVC{
+			Path: path,
+			Dir:  dir,
+		},
+	}
+	builder.globalFlags = f
+	return builder.cmd()
+}
+
+type cmdConfigBuilder struct {
+	genericCLIOpts
+	*globalFlags
+
+	name   string
+	url    string
+	token  string
+	active bool
+	org    string
+
+	svc config.ConfigsService
+}
+
+func (b *cmdConfigBuilder) cmd() *cobra.Command {
+	cmd := b.newCmd("config", nil)
+	cmd.Short = "Config management commands"
+	cmd.Run = seeHelp
+	cmd.AddCommand(
+		b.cmdCreate(),
+		b.cmdDelete(),
+		b.cmdUpdate(),
+		b.cmdList(),
+	)
+	return cmd
+}
+
+func (b *cmdConfigBuilder) cmdCreate() *cobra.Command {
+	cmd := b.newCmd("create", b.cmdCreateRunEFn)
+	cmd.Short = "Create config"
+	cmd.Flags().StringVarP(&b.name, "name", "n", "", "The config name (required)")
+	cmd.MarkFlagRequired("name")
+	cmd.Flags().StringVarP(&b.token, "token", "t", "", "The config token (required)")
+	cmd.MarkFlagRequired("token")
+	cmd.Flags().StringVarP(&b.url, "url", "u", "", "The config url (required)")
+	cmd.MarkFlagRequired("url")
+
+	cmd.Flags().BoolVarP(&b.active, "active", "a", false, "Set it to be the active config")
+	cmd.Flags().StringVarP(&b.org, "org", "o", "", "The optional organization name")
+	return cmd
+}
+
+func (b *cmdConfigBuilder) cmdCreateRunEFn(*cobra.Command, []string) error {
+	pp, err := b.svc.ParseConfigs()
+	if err != nil {
+		return err
+	}
+	p := config.Config{
+		Host:   b.url,
+		Token:  b.token,
+		Org:    b.org,
+		Active: b.active,
+	}
+	if _, ok := pp[b.name]; ok {
+		return &influxdb.Error{
+			Code: influxdb.EConflict,
+			Msg:  fmt.Sprintf("name %q already exists", b.name),
+		}
+	}
+	pp[b.name] = p
+	if p.Active {
+		if err := pp.Switch(b.name); err != nil {
+			return err
+		}
+	}
+	return b.svc.WriteConfigs(pp)
+}
+
+func (b *cmdConfigBuilder) cmdDelete() *cobra.Command {
+	cmd := b.newCmd("delete", b.cmdDeleteRunEFn)
+	cmd.Short = "Delete config"
+
+	cmd.Flags().StringVarP(&b.name, "name", "n", "", "The config name (required)")
+	cmd.MarkFlagRequired("name")
+
+	return cmd
+}
+
+func (b *cmdConfigBuilder) cmdDeleteRunEFn(cmd *cobra.Command, args []string) error {
+	pp, err := b.svc.ParseConfigs()
+	if err != nil {
+		return err
+	}
+	if _, ok := pp[b.name]; !ok {
+		return &influxdb.Error{
+			Code: influxdb.ENotFound,
+			Msg:  fmt.Sprintf("name %q is not found", b.name),
+		}
+	}
+	delete(pp, b.name)
+	return b.svc.WriteConfigs(pp)
+}
+
+func (b *cmdConfigBuilder) cmdUpdate() *cobra.Command {
+	cmd := b.newCmd("set", b.cmdUpdateRunEFn)
+	cmd.Short = "Update config"
+	cmd.Flags().StringVarP(&b.name, "name", "n", "", "The config name (required)")
+	cmd.MarkFlagRequired("name")
+
+	cmd.Flags().StringVarP(&b.token, "token", "t", "", "The config token (required)")
```

Comment:
the global vs local `token` flag feels weird still, we need to figure out how to remedy this 🤔

Reply:
agree, but maybe later.